mono/metadata/object-internals.h

```diff
@@ -1941,8 +1941,24 @@ mono_string_to_utf8_ignore (MonoString *s);
 gboolean
 mono_monitor_is_il_fastpath_wrapper (MonoMethod *method);
 
-MonoString*
-mono_string_intern_checked (MonoString *str, MonoError *error);
+MonoStringHandle
+mono_string_is_interned_lookup (MonoStringHandle str, gboolean insert, MonoError *error);
+
+/**
+ * mono_string_intern_checked:
+ * \param str String to intern
+ * \param error set on error.
+ * Interns the string passed.
+ * \returns The interned string. On failure returns NULL and sets \p error
+ */
+#define mono_string_intern_checked(str, error) (mono_string_is_interned_lookup ((str), TRUE, (error)))
+
+/**
+ * mono_string_is_interned_internal:
+ * \param o String to probe
+ * \returns Whether the string has been interned.
+ */
+#define mono_string_is_interned_internal(str, error) (mono_string_is_interned_lookup ((str), FALSE, (error)))
```

Comment:
shouldn't this be
```suggestion
 #define mono_string_is_interned_internal(str, error) (MONO_HANDLE_BOOL (mono_string_is_interned_lookup ((str), FALSE, (error))))
```

Reply:
And  `ves_icall_System_String_InternalIsInterned`, at least first order.

Reply:
I agree is is confusing but this fits the prior pattern and the callers' prior and remaining requirements. Maybe rename somehow?

Reply:
I agree is is confusing but this fits the prior pattern and the callers' prior and remaining requirements. Maybe rename somehow?